kolibri/plugins/facility_management/assets/src/views/UserPage/index.vue

```diff
@@ -166,59 +140,31 @@
           { label: this.$tr('admins'), value: UserKinds.ADMIN },
         ];
       },
-      sortedFilteredUsers() {
-        return filterAndSortUsers(
-          this.facilityUsers,
-          user => userMatchesFilter(user, this.searchFilter) && this.userMatchesRole(user)
-        );
-      },
-      visibleUsers() {
-        return this.sortedFilteredUsers.slice(this.startRange, this.endRange);
-      },
-      emptyMessage() {
-        if (this.facilityUsers.length === 0) {
-          return this.coreString('noUsersExistLabel');
-        } else if (this.visibleUsers.length === 0) {
-          return this.$tr('allUsersFilteredOut');
-        }
-        return '';
-      },
-      numPages() {
-        return Math.ceil(this.numFilteredUsers / this.perPage);
-      },
-      startRange() {
-        return (this.pageNum - 1) * this.perPage;
-      },
-      visibleStartRange() {
-        return Math.min(this.startRange + 1, this.numFilteredUsers);
-      },
-      endRange() {
-        return this.pageNum * this.perPage;
-      },
-      visibleEndRange() {
-        return Math.min(this.endRange, this.numFilteredUsers);
-      },
-      numFilteredUsers() {
-        return this.sortedFilteredUsers.length;
-      },
-    },
-    watch: {
-      searchFilter() {
-        // Reset the pageNum to the first page when searchFilter changes
-        // to avoid showing an empty page.
-        this.pageNum = 1;
-      },
     },
     beforeMount() {
       this.roleFilter = this.userKinds[0];
     },
     methods: {
       ...mapActions('userManagement', ['displayModal']),
+      emptyMessageForItems(items, filterText) {
+        if (this.facilityUsers.length === 0) {
+          return this.$tr('noUsersExist');
+        } else if (items.length === 0) {
+          return this.$tr('allUsersFilteredOut', { filterText });
+        }
+        return '';
+      },
+      filterUsers(users, filterText, { roleFilter }) {
```

Comment:
Why does this need to be passed into the function, rather than referenced as `this.roleFilter` within the function?

I feel like this would simplify the API, and would mean that you only need to reset the page number when the `filteredItems` changes in the `PaginatedListComponent`.

Reply:
I'm not sure anymore. My guess is that I needed pass `otherFilters` into the PaginatedListContainer to make its `filteredItems` element reactive to changes in `otherFilters` as well as the default textbox input.

Reply:
that approach actually does work, but uses a little advanced JS (having to `.bind(this)` the reactive function).

Reply:
Hrm, I guess my hope was that because `filterUsers` is a method of the component, its `this` would remain bound to the component itself, and then using it inside the computed prop of the `PaginatedListComponent` would access the reactive properties on the `UserPage` component, and then that would trigger recomputation on change.